rust/arrow/src/csv/reader.rs

```diff
@@ -446,8 +446,57 @@ fn parse(
     arrays.and_then(|arr| RecordBatch::try_new(projected_schema, arr))
 }
 
+trait Parser: ArrowPrimitiveType {
+    fn parse(string: &str) -> Option<Self::Native> {
+        string.parse::<Self::Native>().ok()
+    }
+}
+
+impl Parser for BooleanType {
+    fn parse(string: &str) -> Option<bool> {
```

Comment:
Added https://issues.apache.org/jira/browse/ARROW-10677 to track extra parsing

Reply:
Yes, the mixed case is to keep compatibility with the previous implementation.

Could as well have specific cases for all caps / capitalized booleans instead maybe? That would avoid accepting tRue etc.

Reply:
I think the code in this PR is good as is -- I just think it would be nice to document what the behavior is more explicitly -- so I did  so in https://github.com/apache/arrow/pull/8733